src/contract.rs

```diff
@@ -114,12 +129,19 @@ impl<T: Transport> Instance<T> {
     /// Returns a method builder to setup a call or transaction on a smart
     /// contract method. Note that calls just get evaluated on a node but do not
     /// actually commit anything to the block chain.
-    pub fn method<S, P, R>(&self, name: S, params: P) -> AbiResult<MethodBuilder<T, R>>
+    pub fn method<S, P, R>(&self, signature: S, params: P) -> AbiResult<MethodBuilder<T, R>>
     where
         S: AsRef<str>,
         P: Tokenize,
     {
-        let function = self.abi.function(name.as_ref())?;
+        // NOTE: First try to get the method by signature, and if that fails
+        //   then by name.
+        let signature = signature.as_ref();
+        let function = self
+            .methods
+            .get(signature)
+            .map(|(name, index)| Ok(&self.abi.functions[name][*index]))
+            .unwrap_or_else(|| self.abi.function(signature))?;
```

Comment:
In which case is this possible? If I understand correctly we always pass in the signature literal now, so if it doesn't exist in the Map you created it won't exist at all, would it? `abi.function` seems to work on method names not signatures.

Reply:
This is just for backwards compatibility in case someone was using the old `Instance` API. Now that you brought it up, it may not be worth supporting it.